x/profiles/types/models_chain_links.go

```diff
@@ -87,19 +90,29 @@ func (p Proof) Validate() error {
 
 // Verify verifies the signature using the given plain text and public key.
 // It returns and error if something is invalid.
-func (p Proof) Verify(unpacker codectypes.AnyUnpacker, address AddressData) error {
+func (p Proof) Verify(cdc codec.BinaryCodec, legacyAmino *codec.LegacyAmino, address AddressData) error {
 	value, err := hex.DecodeString(p.PlainText)
 	if err != nil {
 		return fmt.Errorf("error while decoding proof text: %s", err)
 	}
 
+	// Make sure the signed value is valid, if it's a transaction
+	isValidTextSig := IsValidTextSig(value, address.GetValue())
```

Comment:
We should check the target address is memo or not, instead of the source one.

Reply:
You're right, thanks! Updated this